### PR TITLE
[s2geometry] update code & fix windows builds

### PR DESCRIPTION
**Pull request by jherico**
- [X] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/contributing/maintainer-guide.md)
- [X] SHA512s are updated for each updated download
- [X] The "supports" clause reflects platforms that may be fixed by this new version
- [X] Any fixed [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt) entries are removed from that file.
- [X] Any patches that are no longer applied are deleted from the port's directory.
- [X] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
- [X] Only one version is added to each modified port's versions file.

I was able to get a [PR](https://github.com/google/s2geometry/pull/338) accepted upstream that fixes the issues preventing a builds on Windows (or more specifically MSVC) and have an open [PR](https://github.com/google/s2geometry/pull/339) adding proper CMake config targets similar to the current portfile patch, but I'm uncertain whether the latter will get merged in the near term.  As such I'm updating the current port code to target the fixes with the existing (updated) patches in order to unblock another new port I'd like to get added to vcpkg in the near future, but which depends on s2geometry.

### CODE REVIEW

File: ports/s2geometry/vcpkg.json [hunk 1]
```diff
@@ -1,14 +1,12 @@
 {
   "name": "s2geometry",
   "version": "0.10.0",
+  "port-version": 1,
```
Comment (LilyWangLL) [inline ports/s2geometry/vcpkg.json:4]:
```suggestion
  "version-date": "2024-01-13",
```

**Review by LilyWangLL — CHANGES_REQUESTED** [review]

File: ports/s2geometry/vcpkg.json (hunk 1 above)
Comment (LilyWangLL) [reply ports/s2geometry/vcpkg.json:4]: This is port update, please remove `port-version` and use `version-date` instead of `version`.

**Review by LilyWangLL — DISMISSED** [review]

**Comment by JavierMatosD** [comment]
While I appreciate the intent to integrate upstream fixes, diverging from the official release schedule and changing the versioning scheme is not typically recommended in vcpkg. Different versioning schemes can lead to difficulties in version comparison and dependency management.

If the purpose is to include specific fixes that have been merged upstream, a more appropriate approach would be to apply those patches directly to this port. This way, we can incorporate necessary updates without deviating from the established upstream release versions.

Please ensure any applied patches are well-documented in the port file, including details about their origin and necessity. We will update the port to the new official release when it becomes available from the upstream project.

**Comment by jherico** [comment]
OK, well, I guess I'll ask them to cut a release and see what they say.  https://github.com/google/s2geometry/issues/341

**Comment by jherico** [comment]
this is waiting on the upstream repo to add the tag.  once it does the CI should pass.

**Review by LilyWangLL — APPROVED** [review]

File: ports/s2geometry/usage [hunk 2]
```diff
@@ -1,4 +1,4 @@
 The package s2geometry provides CMake targets:
 
-    find_package(unofficial-s2 CONFIG REQUIRED)
-    target_link_libraries(main PRIVATE unofficial-s2::s2)
+    find_package(s2 CONFIG REQUIRED)
+    target_link_libraries(main PRIVATE s2::s2)
```
Comment (LilyWangLL) [inline ports/s2geometry/usage:4]:
```suggestion
    target_link_libraries(main PRIVATE s2s2)
```

**Review by LilyWangLL — CHANGES_REQUESTED** [review]

**Comment by LilyWangLL** [comment]
And please add the following codes to `Config.cmake.in`.
```
include(CMakeFindDependencyMacro)
find_dependency(absl CONFIG)
```

Fix the error:
```
1> [CMake] CMake Error at F:/PRreview/vcpkg/installed/x64-windows/share/s2/s2Targets.cmake:61 (set_target_properties):
1> [CMake]   The link interface of target "s2s2" contains:
1> [CMake] 
1> [CMake]     absl::base
1> [CMake] 
1> [CMake]   but the target was not found.  Possible reasons include:
```

**Comment by LilyWangLL** [comment]
Usage test passed on `x64-windows`.

**Review by LilyWangLL — APPROVED** [review]

**Pull request merged**